echidna/limit/chi_squared.py

```diff
@@ -147,20 +124,23 @@ def pearson_chi_squared(observed, expected):
         events
 
     Raises:
-      ValueError: If either array contains a bin with content <= 0.0
+      ValueError: If arrays are different lengths.
 
     Returns:
       float: Calculated Pearson's chi squared
     """
-    correct_bin_content, message = check_bin_content(observed)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: observed " +
-                         message)
-    correct_bin_content, message = check_bin_content(expected)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: expected " +
-                         message)
-    return numpy.sum((observed-expected)**2 / expected)
+    if len(observed) != len(expected):
+        raise ValueError("Arrays are different lengths")
+    epsilon = 1e-34  # In the limit of zero
+    total = 0
+    for i in range(len(observed)):
```

Comment:
Is there a noticeable difference in performance using a python for loop over the elementwise `numpy.sum` routine. This may be something we need to consider in the future. The use of `numpy.sum` was initially chosen because it is faster than using a python for loop (see comments on #14). This may become a problem as we start to include more and more backgrounds.